src/docfx/lib/github/GitHubUserCache.cs

```diff
@@ -142,22 +158,22 @@ public async Task<Error> SaveChanges(Config config)
                     Users = Users.ToArray(),
```

Comment:
> Users.ToArray() [](start = 27, length = 16)

toArray already makesure it's an snipshoot, no need lock?

Reply:
This also ensures thread safe for every `User` instance, as `UpdateUser` will modify its property.

---
In reply to: [239725394](https://github.com/dotnet/docfx/pull/3740#discussion_r239725394) [](ancestors = 239725394)